mathics/eval/files_io/files.py

```diff
@@ -14,8 +14,14 @@
 from mathics.core.read import MathicsOpen
 from mathics.core.symbols import SymbolNull
 from mathics.core.systemsymbols import SymbolFailed, SymbolPath
+from mathics.core.util import canonic_filename
 
-# Python representation of $InputFileName
+# Python representation of $InputFileName.  On Windows platforms, we
+# canonicalize this to its Posix equvivalent name.
+# FIXME: Remove this as a module-level variable and instead
+#        define it in a session definitions object.
+#        With this, multiple sessions will have separate
+#        $InputFilename
```

Comment:
This goes back to https://github.com/Mathics3/mathics-core/pull/1012#pullrequestreview-1888135198 and somewhere else that I can't find where you said this could be done later. 

I think things would have to be reworked so that insteads of getting this from a module variable it would only be available from some place accessible from a session object.